File: kaftter-consumer/src/main/java/kaftter/consumer/converter/TweetConverter.java
package kaftter.consumer.converter;

import kaftter.tweet.Tweet;
import kaftter.consumer.vo.User;
import lombok.experimental.UtilityClass;

@UtilityClass
public class TweetConverter {
    public kaftter.consumer.vo.Tweet convert(final Tweet tweet) {
        final var user = User.builder()
                .id(tweet.getUser().getId())
                .name(tweet.getUser().getName())
                .screenName(tweet.getUser().getScreenName())
                .followers(tweet.getUser().getFollowers())
                .build();
        return kaftter.consumer.vo.Tweet.builder()
                .id(tweet.getId())
                .text(tweet.getText())
                .user(user)
                .quoteCount(tweet.getQuoteCount())
                .favoriteCount(tweet.getFavoriteCount())
                .replyCount(tweet.getReplyCount())
                .retweetCount(tweet.getRetweetCount())
                .language(tweet.getLanguage())
                .build();
    }
}

File: kaftter-api/src/main/kotlin/kaftter/api/converter/TweetConverter.kt
package kaftter.api.converter

import kaftter.api.domain.SummarizedTweetEntity
import kaftter.api.domain.TweetEntity
import kaftter.api.domain.TweetKey
import kaftter.api.vo.Tweet
import kaftter.api.vo.TweetSummary
import java.time.LocalDate

fun convert(summarizedTweetEntity: SummarizedTweetEntity): TweetSummary {
    return TweetSummary(
            userId = summarizedTweetEntity.userId,
            userName = summarizedTweetEntity.userName,
            favoriteCount = summarizedTweetEntity.favoriteCount,
            quoteCount = summarizedTweetEntity.quoteCount,
            replyCount = summarizedTweetEntity.replyCount,
            retweetCount = summarizedTweetEntity.retweetCount
    )
}

fun convert(tweet: Tweet): TweetEntity {
    val tweetKey = TweetKey(
            tweetId = tweet.id,
            userId = tweet.user.id,
            createdAt = LocalDate.now()
    )
    return TweetEntity(
            key = tweetKey,
            text = tweet.text,
            quoteCount = tweet.quoteCount,
            replyCount = tweet.replyCount,
            retweetCount = tweet.retweetCount,
            favoriteCount = tweet.favoriteCount,
            language = tweet.language,
            userName = tweet.user.name,
            userScreenName = tweet.user.screenName,
            userFollowers = tweet.user.followers
    )
}
